duckduckgo/iOS
# Update Overlay Pixel implementation

Task/Issue URL: https://app.asana.com/0/1204099484721401/1208910025199824/f
Tech Design URL:
CC:

**Description**:
Refactors Overlay Pixels to use WebKit KVO and backForward lists for improved reliability

**Steps to test this PR**:
1. Filter the console messages for "Pixel fired."
2. Go to Youtube.com
3. Search for MKBHD
4. Select a video
5. Hit Back
6. Confirm  ` duckplayer.youtube.overlay.navigation.back` pixel fires
7. Select another video
8. Hit Refresh
9. Confirm `duckplayer.youtube.overlay.navigation.refresh`pixel fires
10. Navigate to other Youtube video suggestion
11. Confirm `duckplayer.youtube.overlay.navigation.within-youtube`
12. Visit another website (i.e. example.com)
13. Confirm `duckplayer.youtube.overlay.navigation.outside-youtube` fires

Smoke test going to/from videos and confirm pixels fire as expected.
